internal/circle/service.go

```diff
@@ -51,6 +54,10 @@ func (o ServiceOptions) Validate() error {
 		return fmt.Errorf("TenantManager is required")
 	}
 
+	if o.MonitorService == nil {
+		return fmt.Errorf("MonitorService is required")
+	}
+
```

Comment:
Should monitoring be mandatory?

Reply:
We discussed this in DMs. Metrics are always collected, (e.g. `sql_exec_with_metrics`) but they can be published or not, based on the flag `METRICS_ENABLED`.